code/src/java/pcgen/gui2/csheet/CharacterSheetPanel.java

```diff
@@ -56,61 +60,36 @@
  */
 public class CharacterSheetPanel extends HtmlPanel implements CharacterSelectionListener
 {
-
-	private static final String COLOR_TAG = "preview_color.css";
-
 	private enum CssColor
 	{
-
-		BLUE,
-		LIGHTBLUE,
-		GREEN,
-		LIGHTGREEN,
-		RED,
-		LIGHTRED,
-		YELLOW,
-		LIGHTYELLOW,
-		GREY,
-		LIGHTGREY;
-
-		public String getCssText()
-		{
-			switch (this)
-			{
-				case BLUE:
-					return "preview_color_blue.css";
-				case LIGHTBLUE:
-					return "preview_color_light_blue.css";
-				case GREEN:
-					return "preview_color_green.css";
-				case LIGHTGREEN:
-					return "preview_color_light_green.css";
-				case RED:
-					return "preview_color_red.css";
-				case LIGHTRED:
-					return "preview_color_light_red.css";
-				case YELLOW:
-					return "preview_color_yellow.css";
-				case LIGHTYELLOW:
-					return "preview_color_light_yellow.css";
-				case GREY:
-					return "preview_color_grey.css";
-				case LIGHTGREY:
-					return "preview_color_light_grey.css";
-				default:
-					throw new InternalError();
-			}
+		BLUE("preview_color_blue.css"),
+		LIGHTBLUE("preview_color_light_blue.css"),
+		GREEN("preview_color_green.css"),
+		LIGHTGREEN("preview_color_light_green.css"),
+		RED("preview_color_red.css"),
+		LIGHTRED("preview_color_light_red.css"),
+		YELLOW("preview_color_yellow.css"),
+		LIGHTYELLOW("preview_color_light_yellow.css"),
+		GREY("preview_color_grey.css"),
+		LIGHTGREY("preview_color_light_grey.css");
+
+		public String cssfile;
```

Comment:
Abstraction:

Abstraction in java can happen by using “abstract classes” and
“interfaces”. An abstract class or interface is common ideas that are
used by other objects. In java terms, abstract classes are ‘extended’ by
concrete classes. As an example, a dog (concrete class) can be said to
be a canine (abstract), and then furthermore an animal (abstract). The
java GUI classes are a great example of abstract classes.
Conversationally, an abstract class must pass the IS-A test for
inheritance. A canine IS-A animal, a dog IS-A canine. The other
mentioned abstract is interfaces. An interface does not need to pass the
IS-A test. I loved the example of Robot and Pet being interfaces. A dog
could be a Robot, and it can be a Pet, but not all dogs are a Pet, nor
are they all Robots. An interface is basically a set of ideas that can
be used by an object. The big point I learned is classes may only have
one abstract class above them, but may have as many interfaces as they want.


Not exactly sure how by following good coding practices of "hiding data"
that violates the principles of abstraction (common ideas). And since
this is literally the discussion for this week, I'm fairly certain I
have those basics down.

Cheers,

On 10/1/2015 6:59 PM, grimreaper wrote:
>
> In code/src/java/pcgen/gui2/csheet/CharacterSheetPanel.java
> <https://github.com/PCGen/pcgen/pull/639#discussion_r40986360>:
>
> > -					return "preview_color_light_grey.css";
> > -				default:
> > -					throw new InternalError();
> > -			}
> > +		BLUE("preview_color_blue.css"),
> > +		LIGHTBLUE("preview_color_light_blue.css"),
> > +		GREEN("preview_color_green.css"),
> > +		LIGHTGREEN("preview_color_light_green.css"),
> > +		RED("preview_color_red.css"),
> > +		LIGHTRED("preview_color_light_red.css"),
> > +		YELLOW("preview_color_yellow.css"),
> > +		LIGHTYELLOW("preview_color_light_yellow.css"),
> > +		GREY("preview_color_grey.css"),
> > +		LIGHTGREY("preview_color_light_grey.css");
> > +
> > +		public String cssfile;
>
> I much prefer using final public. Using getters / setters is a
> violation of abstraction and can make code harder to reason about.
> that said, I'll go with whatever the project style dictates
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PCGen/pcgen/pull/639/files#r40986360>.
>